### PR TITLE
Create new state machine for renewals

### PR DESCRIPTION
**Pull request by irisfaraway**
https://eaflood.atlassian.net/browse/RUBY-516

This story covers:

- creating a new state machine to handle the renewal workflow
- applying the renewal state machine to the renewals model
- creating skeleton versions of the ‘Do you want to renew with these details?', 'you are about to renew', and 'renewal complete’ pages - each page should just be a heading and a 'continue' button

### CODE REVIEW

**Comment by irisfaraway** [comment]
For the sake of keeping our PRs relatively small, I'm going to handle putting the existing forms into the renewal workflow separately.

**Comment by irisfaraway** [comment]
Rebased and ready to review!

File: app/models/concerns/waste_exemptions_engine/can_use_renewing_registration_workflow.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+module WasteExemptionsEngine
+  module CanUseRenewingRegistrationWorkflow
+    extend ActiveSupport::Concern
+
+    included do
+      include AASM
+
+      aasm column: :workflow_state do
+        # States / forms
+
```
Comment (cintamani) [inline app/models/concerns/waste_exemptions_engine/can_use_renewing_registration_workflow.rb]: Suoper NIT
```suggestion
```

File: app/controllers/waste_exemptions_engine/renewal_complete_forms_controller.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module WasteExemptionsEngine
+  class RenewalCompleteFormsController < FormsController
+    def new
+      return unless super(RenewalCompleteForm, "renewal_complete_form")
+    end
+
+    # Overwrite create and go_back as you shouldn't be able to submit or go back
```
[the remark this answers is not in the document]
Comment (irisfaraway) [reply app/controllers/waste_exemptions_engine/renewal_complete_forms_controller.rb]: Not in this case - this covers the state-machine-approved actions of going forward and back, as well as calling submit on the form object.

"Flexible navigation" is about skipping to pages by using the browser back button or editing the URL – anything that goes around the AASM workflow.

[the remark this answers is not in the document]
Comment (cintamani) [reply app/controllers/waste_exemptions_engine/renewal_complete_forms_controller.rb]: Makes sense! :) 
Still, we have this overrides in many other places, shall we create a concern to handle these cases? I think is worth creating a housekeeping issue so we can tackle it later :)

[the remark this answers is not in the document]
Comment (irisfaraway) [reply app/controllers/waste_exemptions_engine/renewal_complete_forms_controller.rb]: OK, I'll get something added!

**Pull request merged**